quantization/src/encoded_vectors_binary.rs

```diff
@@ -120,30 +121,38 @@ impl<TStorage: EncodedStorage> EncodedVectorsBin<TStorage> {
     }
 
     fn calculate_metric(&self, v1: &[BitsStoreType], v2: &[BitsStoreType]) -> f32 {
-        let xor_product = Self::xor_product(v1, v2);
-
         // Dot product in a range [-1; 1] is approximated by NXOR in a range [0; 1]
+        // L1 distance in range [-1; 1] (alpha=2) is approximated by alpha*XOR in a range [0; 1]
+        // L2 distance in range [-1; 1] (alpha=2) is approximated by alpha*sqrt(XOR) in a range [0; 1]
         // For example:
 
-        // A    |   B   | Dot product
-        // -0.5 | -0.5  |  0.25
-        // -0.5 |  0.5  | -0.25
-        //  0.5 | -0.5  | -0.25
-        //  0.5 |  0.5  |  0.25
-
-        // A | B  |  NXOR
-        // 0 | 0  |  1
-        // 0 | 1  |  0
-        // 1 | 0  |  0
-        // 1 | 1  |  1
-
-        // So is `invert` is true, we return XOR, otherwise we return (dim - XOR)
-
-        let zeros_count = self.metadata.vector_parameters.dim - xor_product;
-        if self.metadata.vector_parameters.invert {
-            xor_product as f32 - zeros_count as f32
-        } else {
-            zeros_count as f32 - xor_product as f32
+        // |  A   |  B   | Dot product | L1 | L2 |
+        // | -0.5 | -0.5 |  0.25       | 0  | 0  |
+        // | -0.5 |  0.5 | -0.25       | 1  | 1  |
+        // |  0.5 | -0.5 | -0.25       | 1  | 1  |
+        // |  0.5 |  0.5 |  0.25       | 0  | 0  |
+
+        // | A | B | NXOR | XOR
+        // | 0 | 0 | 1    | 0
+        // | 0 | 1 | 0    | 1
+        // | 1 | 0 | 0    | 1
+        // | 1 | 1 | 1    | 0
+
+        let xor_product = Self::xor_product(v1, v2) as f32;
+
+        let dim = self.metadata.vector_parameters.dim as f32;
+        let zeros_count = dim - xor_product;
+
+        match (
```

Comment:
thanks for fixing `invert` parameter! It seems you may do this part of code simpler because with `invert=true` you can just multiply result to `-1`

Reply:
I like my overly explicit way more actually, I want to keep it if that's OK for you? 

It shows that Dot product is a similarity metric but L1 and L2 are distances that needs to be inverted to become a similarity metrics.

Reply:
OK for me